netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1297,103 +1318,43 @@ net_ebpf_extension_sock_addr_redirect_connection_classify(
     NTSTATUS status = STATUS_SUCCESS;
     net_ebpf_extension_sock_addr_wfp_filter_context_t* filter_context = NULL;
     net_ebpf_extension_hook_client_t* attached_client = NULL;
-    net_ebpf_sock_addr_t* net_ebpf_sock_addr_ctx = NULL;
-    bpf_sock_addr_t* sock_addr_ctx = NULL;
-    bpf_sock_addr_t* sock_addr_ctx_original = NULL;
+    net_ebpf_sock_addr_t net_ebpf_sock_addr_ctx = {0};
+    bpf_sock_addr_t* sock_addr_ctx = (bpf_sock_addr_t*)&net_ebpf_sock_addr_ctx.base;
+    bpf_sock_addr_t sock_addr_ctx_original = {0};
     uint32_t compartment_id = UNSPECIFIED_COMPARTMENT_ID;
+    bool v4_mapped = FALSE;
     FWPS_CONNECTION_REDIRECT_STATE redirect_state;
     HANDLE redirect_handle;
     uint64_t classify_handle = 0;
-    net_ebpf_extension_connection_context_t* connection_context_original = NULL;
-    net_ebpf_extension_connection_context_t* connection_context_redirected = NULL;
-    BOOLEAN redirected = FALSE;
-    FWP_ACTION_TYPE action = FWP_ACTION_PERMIT;
-    BOOLEAN classify_handle_acquired = FALSE;
-    BOOLEAN v4_mapped = FALSE;
-    BOOLEAN is_original_connection;
+    bool classify_handle_acquired = FALSE;
+    bool redirected = FALSE;
+    net_ebpf_extension_connection_context_t* blocked_connection_context = NULL;
 
     UNREFERENCED_PARAMETER(layer_data);
     UNREFERENCED_PARAMETER(flow_context);
 
     if ((classify_output->rights & FWPS_RIGHT_ACTION_WRITE) == 0) {
-        // Do not modify anything and bail.
+        // A callout with higher weight has revoked the write permission. Bail out.
```

Comment:
Was wondering why this is not logged? If this is a benign 'noise generator', a brief comment to that effect might be helpful from future maintenance perspective, as the existing comment give a different impression.